core/src/test/java/org/apache/iceberg/TestMetadataTableScansWithPartitionEvolution.java

```diff
@@ -222,6 +222,14 @@ public void testPositionDeletesPartitionSpecRemoval() {
         constantsMap(posDeleteTask, partitionType).get(MetadataColumns.FILE_PATH.fieldId()));
   }
 
+  @Test
+  public void testPartitionSpecEvolutionToUnpartitioned() {
```

Comment:
feels like test name is not super clear, maybe `testPreviouslyPartitionedSpecStillShowPartitionColumnWhenEvolveToUnpartitioned` ?

Reply:
Hm. it looks a bit long :)  In that case, Id vote to maybe bring back the original asserts then the first commit so its a more end-to-end test , https://github.com/apache/iceberg/pull/7551/commits/54b95ea8658d7db03eea717e68b09350fff13a9a  as I dont think its covered elsewhere (though realize its not related to this pr )